jib-core/src/main/java/com/google/cloud/tools/jib/event/EventHandlers.java

```diff
@@ -24,45 +25,90 @@
+  public static class Builder {
+
+    private final Multimap<Class<? extends JibEvent>, Handler<? extends JibEvent>> handlers =
+        ArrayListMultimap.create();
+
+    /**
+     * Adds the {@code eventConsumer} to handle the {@link JibEvent} with class {@code eventClass}.
+     * The order in which handlers are added is the order in which they are called when the event is
+     * dispatched.
+     *
+     * <p><b>Note: Implementations of {@code eventConsumer} must be thread-safe.</b>
+     *
+     * @param eventType the event type that {@code eventConsumer} should handle
+     * @param eventConsumer the event handler
+     * @param <E> the type of {@code eventClass}
+     * @return this
+     */
+    public <E extends JibEvent> Builder add(JibEventType<E> eventType, Consumer<E> eventConsumer) {
+      Class<E> eventClass = eventType.getEventClass();
+      handlers.put(eventClass, new Handler<>(eventClass, eventConsumer));
+      return this;
+    }
+
+    /**
+     * Adds the {@code eventConsumer} to handle all {@link JibEvent} types. See {@link
+     * #add(JibEventType, Consumer)} for more details.
+     *
+     * @param eventConsumer the event handler
+     * @return this
+     */
+    public Builder add(Consumer<JibEvent> eventConsumer) {
+      return add(JibEventType.ALL, eventConsumer);
+    }
+
+    public EventHandlers build() {
+      return new EventHandlers(handlers);
+    }
+  }
+
   /** Maps from {@link JibEvent} class to handlers for that event type. */
-  private final Multimap<Class<? extends JibEvent>, Handler<? extends JibEvent>> handlers =
-      ArrayListMultimap.create();
+  private final ImmutableMultimap<Class<? extends JibEvent>, Handler<? extends JibEvent>> handlers;
+
+  private EventHandlers(Multimap<Class<? extends JibEvent>, Handler<? extends JibEvent>> handlers) {
+    this.handlers = ImmutableMultimap.copyOf(handlers);
+  }
 
   /**
-   * Adds the {@code eventConsumer} to handle the {@link JibEvent} with class {@code eventClass}.
-   * The order in which handlers are added is the order in which they are called when the event is
-   * dispatched.
+   * Returns an empty {@link EventHandlers}.
    *
-   * <p><b>Note: Implementations of {@code eventConsumer} must be thread-safe.</b>
+   * @return the {@link EventHandlers}
+   */
+  public static EventHandlers none() {
+    return new Builder().build();
+  }
+
+  /**
+   * Creates a new {@link EventHandlers.Builder}.
    *
-   * @param eventType the event type that {@code eventConsumer} should handle
-   * @param eventConsumer the event handler
-   * @param <E> the type of {@code eventClass}
-   * @return this
+   * @return the builder
    */
-  public <E extends JibEvent> EventHandlers add(
-      JibEventType<E> eventType, Consumer<E> eventConsumer) {
-    Class<E> eventClass = eventType.getEventClass();
-    handlers.put(eventClass, new Handler<>(eventClass, eventConsumer));
-    return this;
+  public static Builder builder() {
+    return new Builder();
```

Comment:
Personally I don't see a great value of this method, since I can do `new EventHandlers.Builder()`. Is this considered a good practice? I don't know very well.

Reply:
I'm not sure either, but it's consistent with the rest of the code base (and is a bit more concise).

Reply:
OK, we're good then.